### PR TITLE
Alarm node changed to read alarm config from message

### PR DESCRIPTION
**Pull request by mp-loki**

### CODE REVIEW

File: rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java [hunk 1]
```diff
@@ -49,29 +52,55 @@
 )
 public class TbCreateAlarmNode extends TbAbstractAlarmNode<TbCreateAlarmNodeConfiguration> {
 
+    private ObjectMapper mapper = new ObjectMapper();
```
Comment (ashvayka) [inline rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java]: I believe you can use static ObjectMapper: https://stackoverflow.com/questions/3907929/should-i-declare-jacksons-objectmapper-as-a-static-field

File: rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java [hunk 2]
```diff
@@ -49,29 +52,55 @@
...
     }
 
-    private ListenableFuture<AlarmResult> createNewAlarm(TbContext ctx, TbMsg msg) {
-        ListenableFuture<Alarm> asyncAlarm = Futures.transform(buildAlarmDetails(ctx, msg, null),
-                details -> buildAlarm(msg, details, ctx.getTenantId()));
+    private ListenableFuture<AlarmResult> createNewAlarm(TbContext ctx, TbMsg msg, Alarm msgAlarm) {
+        ListenableFuture<Alarm> asyncAlarm;
+        if (msgAlarm != null ) {
+            asyncAlarm = Futures.immediateCheckedFuture(msgAlarm);
+
```
Comment (ashvayka) [inline rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java]: extra space

File: rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java [hunk 3]
```diff
@@ -49,29 +52,55 @@
...
+                msgAlarm.setTenantId(ctx.getTenantId());
+                alarmType = msgAlarm.getType();
+            } catch (IOException e) {
+                ctx.tellFailure(msg, e);
+                return null;
+            }
+        }
+
+        ListenableFuture<Alarm> latest = ctx.getAlarmService().findLatestByOriginatorAndType(ctx.getTenantId(), msg.getOriginator(), alarmType);
         return Futures.transformAsync(latest, a -> {
```
Comment (ashvayka) [inline rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java]: maybe rename "a" to "dbAlarm" or "existingAlarm" to simplify the code?

**Comment by ashvayka** [comment]
@mp-loki Few minor comments otherwise it is ok!

File: rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java (hunk 3 above)
Comment (mp-loki) [reply rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java]: Requested changes made

**Pull request merged**